objfile/objfile.go

```diff
@@ -182,18 +182,23 @@ func (x byAddr) Swap(i, j int)      { x[i], x[j] = x[j], x[i] }
 
 func findAllOccurrences(data []byte, searches [][]byte) []int {
 	var results []int
+
 	for _, search := range searches {
-		for idx := range data {
-			if len(data[idx:]) < len(search) {
-				continue
-			}
+		var offset = 0
 
-			haystack := data[idx : idx+len(search)]
-			if bytes.Equal(haystack, search) {
-				results = append(results, idx)
+		for {
+			var index = bytes.Index(data[offset:], search)
+			if index == -1 {
+				break
 			}
+
+			offset += index
+			results = append(results, offset)
+
+			offset += 1
```

Comment:
I think it should be ok remove this plus 1 and instead change the above increment "offset += index" into "+= index+len(search)". I cannot think of a case where a match would overlap with another match and should make striding along ever so slightly faster. What do you think?

Reply:
> I cannot think of a case where a match would overlap

I agree that this seems unlikely, but also, since we can imagine it's possible I figured the most correct algorithm was preferable (the alternative would be to introduce more documentation comments explaining the assumptions). I agree that what your proposed change might be a *tiny* bit faster, but I didn't think it outweighed the readability/assumptions.

Reply:
Given that its only faster in the case of a match, and we only expect one to a few matches per file, the speedup is almost totally negligible and I'd challenge you to demonstrate it in a profiler ;-)

Reply:
Your argument about minimal matches in the first place is especially compelling, sounds good to me!